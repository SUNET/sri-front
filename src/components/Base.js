import React from "react";
import { Route, Switch } from "react-router-dom";

import SearchContainer from "../containers/Search";
import { ContactDetails } from "./Contact";

class Base extends React.Component {
    render() {
        return (
            <main className="container">
                <Switch>
                    <Route path="/contacts" component={SearchContainer} />
                    <Route
                        path="/contact/:contactId"
                        component={ContactDetails}
                    />
                    <Route path="/roles" component={SearchContainer} />
                </Switch>
            </main>
        );
    }
}

export default Base;
