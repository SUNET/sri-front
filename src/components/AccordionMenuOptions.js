import React from "react";
import { Nav } from "react-bootstrap";
import { NavLink } from "react-router-dom";
import ToggleSection, { ToggleHeading, TogglePanel } from "../components/ToggleSection";

import "../style/AccordionMenuOptions.scss";

class AccordionMenuOptions extends React.Component {
    renderHeader(option, index) {
        return (
            <ToggleHeading key={`${index}`}>
                <div className="accordion-menu-options__header">{option.header}</div>
            </ToggleHeading>
        );
    }
    renderCollapseList(subOptions, optionIndex) {
        return (
            <TogglePanel>
                {subOptions.map((subOption, subOpIndex) => {
                    return (
                        <div key={`option_menu-${optionIndex}-${subOpIndex}`} className="accordion-menu-options__link">
                            <Nav.Link
                                as={NavLink}
                                activeClassName="active"
                                to={subOption.route}
                                onClick={() => this.props.onSelectedOption()}
                            >
                                <i className={`accordion-menu-options__link__icon ${subOption.iconClass}`}></i>
                                <span className="accordion-menu-options__link__name">{subOption.name}</span>
                            </Nav.Link>
                        </div>
                    );
                })}
            </TogglePanel>
        );
    }
    render() {
        const { data } = this.props;

        return (
            <div className="accordion-menu-options">
                {data.map((option, optionIndex) => {
                    return (
                        <ToggleSection key={`option_menu-${optionIndex}`}>
                            {this.renderHeader(option, optionIndex)}
                            {this.renderCollapseList(option.subOptions, optionIndex)}
                        </ToggleSection>
                    );
                })}
            </div>
        );
    }
}

export default AccordionMenuOptions;
