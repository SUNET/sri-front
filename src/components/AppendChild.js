import React from 'react';
import { Field } from 'redux-form';
import { withTranslation } from 'react-i18next';
import FieldInput from './FieldInput';

class AppendChild extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      name: '',
      fields: [{ name: '' }],
    };
  }

  componentWillMount() {}

  handleFieldNameChange = (index) => (event) => {
    const newShareholders = this.state.fields.map((field, sindex) => {
      if (index !== sindex) return field;
      return { ...field, name: event.target.value };
    });

    this.setState({ fields: newShareholders });
  };

  handleAddFild = (event) => {
    event.preventDefault();
    this.setState({
      fields: this.state.fields.concat([{ name: '' }]),
    });
  };

  render() {
    let { t } = this.props;
    return (
      <div>
        {this.state.fields.map((field, index) => (
          <Field
            component={FieldInput}
            type="text"
            placeholder={`${field.name}${index + 1}`}
            value={field.name}
            onChange={this.handleFieldNameChange(index)}
          />
        ))}
        <button className="btn btn-add outline" onClick={(e) => this.handleAddField(e)}>
          <span>{t('actions/add-new')}</span>
        </button>
      </div>
    );
  }
}

export default withTranslation()(AppendChild);
