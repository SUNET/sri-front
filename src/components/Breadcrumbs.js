import React from 'react';
import Breadcrumb from 'react-bootstrap/Breadcrumb';
import PropTypes from 'prop-types';
import Routes from '../Routes';

function generatePathsArray(pathString, isDetailsEntity, entityDetails) {
  return pathString.split('/').map((pathElement, index, array) => {
    const mathIdCurrentDetailsInLocationPath = pathString.includes(entityDetails.id);
    const isLastPath = index === array.length - 1;
    const isActive = isLastPath;
    let pathTo = `/${pathElement}`;
    let name = Routes[pathTo];

    if (typeof Routes[pathTo] === 'undefined') {
      pathTo = array.slice(0, index + 1).join('/');
      name = isLastPath && isDetailsEntity && mathIdCurrentDetailsInLocationPath ? entityDetails.name : Routes[pathTo];
    }
    return {
      pathTo,
      name,
      isActive,
    };
  });
}

const Breadcrumbs = ({ history, isDetailsEntity, entityDetails }) => {
  const { pathname } = history.location;
  const routeSteps = generatePathsArray(pathname, isDetailsEntity, entityDetails);
  return (
    <div className="breadcrumbs">
      <Breadcrumb>
        {routeSteps.map(({ pathTo, name, isActive }, index) => {
          return (
            <Breadcrumb.Item active={isActive} key={`${index * Math.random()}`} href={pathTo}>
              {name}
            </Breadcrumb.Item>
          );
        })}
      </Breadcrumb>
    </div>
  );
};

Breadcrumbs.propsTypes = {
  history: PropTypes.object.isRequired,
  isDetailsEntity: PropTypes.bool.isRequired,
  entityDetails: PropTypes.bool.isRequired,
};

export default Breadcrumbs;
