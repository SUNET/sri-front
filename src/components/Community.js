import React from "react";
import { Switch, Route, Redirect } from "react-router-dom";
import { withTranslation } from "react-i18next";

import SearchOrganizationContainer from "../containers/organization/SearchOrganization";
import SearchContactContainer from "../containers/contact/SearchContact";
import SearchGroupContainer from "../containers/group/SearchGroup";
// import SubMenu from "./SubMenu";
// import SubMenuActions from "./SubMenuActions";

import "../style/Footer.scss";

class Community extends React.Component {
    render() {
        return (
            <Switch>
                <Redirect exact from="/community" to="/community/organizations" />
                <Route path="/community/organizations" component={SearchOrganizationContainer} />
                <Route path="/community/contacts" component={SearchContactContainer} />
                <Route path="/community/groups" component={SearchGroupContainer} />
            </Switch>
        );
    }
}

export default withTranslation()(Community);
