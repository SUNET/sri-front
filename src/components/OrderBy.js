import React from 'react';
import { withTranslation } from 'react-i18next';
import PropTypes from 'prop-types';

import FieldSwitch from './FieldSwitch';

import '../style/OrderBy.scss';

export class OrderBy extends React.Component {
  handleOnChangeOrderBy(event) {
    const { changeOrderBy } = this.props;
    if (event.target.checked) {
      changeOrderBy('handle_id_ASC');
    } else {
      changeOrderBy('handle_id_DESC');
    }
  }

  render() {
    const { t } = this.props;
    return (
      <div className="order-by outline">
        <FieldSwitch
          type="toggle-icon"
          icon="angle"
          classNames="icon-right"
          labelChecked={t('filter.newest-first')}
          labelUnChecked={t('filter.latest-first')}
          handleChecked={(e) => this.handleOnChangeOrderBy(e)}
          id="orderBy"
        />
      </div>
    );
  }
}

OrderBy.propTypes = {
  changeOrderBy: PropTypes.func.isRequired,
  t: PropTypes.func.isRequired,
};

export default withTranslation()(OrderBy);
