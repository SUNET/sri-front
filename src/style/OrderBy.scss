@import "../variables.scss";

.order-by {
    display: inline-block;

    &.outline {
        border: 1px solid $secondary-color-20;
        border-radius: $input--border-radius;
        padding: 8px 11px;

        svg {
            color: $complementary-color;
        }
    }
}

.order-col {
    .state {
        label {
            text-indent: 0 !important;

            span {
                font-weight: bold;
            }

            .svg-inline--fa {
                margin-left: 2px;
                vertical-align: bottom;
                font-size: 12px;
                color: $complementary-color;
            }
        }
    }
}
