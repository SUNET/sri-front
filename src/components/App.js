import React, { Component } from "react";
import PropTypes from "prop-types";
import { ConnectedRouter } from "connected-react-router";
import Breadcrumbs from "react-router-dynamic-breadcrumbs";
import { Route, Switch } from "react-router-dom";

import FetchingContext from "../components/FetchingContext";
import SplashContainer from "../containers/Splash";
import NotifyContainer from "../containers/Notify";
import TopHeaderContainer from "../containers/TopHeader";
import BaseContainer from "../containers/Base";
import FooterContainer from "../containers/Footer";

import { Row, Col } from "react-bootstrap";
import SideNavNetwork from "./SideNavNetwork";
import SideNavCommunity from "./SideNavCommunity";
import SubMenuActions from "./SubMenuActions";

import Routes from "../Routes";

import "bootstrap/scss/bootstrap.scss";
import "../style/App.scss";
import "../style/Breadcrumbs.scss";
import "../style/SRIButton.scss";

import { history } from "../store";

class App extends Component {
    render() {
        return (
            <FetchingContext.Provider value={this.props.is_fetching}>
                <div className="App container">
                    <ConnectedRouter history={history}>
                        <Row>
                            <Col className="px-0">
                                <TopHeaderContainer />
                            </Col>
                        </Row>
                        <Row>
                            <SplashContainer />
                            <Switch>
                                <Route path="/network" component={SideNavNetwork} />
                                <Route path="/community" component={SideNavCommunity} />
                            </Switch>
                            <Col sm={10} className="fixed-adaptative">
                                <NotifyContainer />
                                <Row className="mt-4">
                                    <Col>
                                        <Breadcrumbs mappedRoutes={Routes} />
                                    </Col>
                                    <Col className="text-right">
                                        <SubMenuActions />
                                    </Col>
                                </Row>
                                {this.props.is_app_loaded && <BaseContainer />}
                            </Col>
                        </Row>
                        <Row>
                            <Col className="px-0">
                                <FooterContainer />
                            </Col>
                        </Row>
                    </ConnectedRouter>
                </div>
            </FetchingContext.Provider>
        );
    }
}

App.propTypes = {
    is_fetching: PropTypes.bool,
    is_app_loaded: PropTypes.bool,
};

export default App;
