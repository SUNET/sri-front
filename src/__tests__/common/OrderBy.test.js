import React from 'react';
import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { OrderBy } from '../../components/OrderBy';

let OrderByWrapper;
let OrderByComponent;
let InputElement;

const defaultProps = {
  changeOrderBy: jest.fn(),
  t: jest.fn((t) => t),
};

const CreateOrderByWrapper = (props) => {
  const allProps = { ...defaultProps, ...props };
  return (
    <div>
      <OrderBy {...allProps} />
    </div>
  );
};

const mountOrderByWrapper = (props) => mount(<CreateOrderByWrapper {...props} />);

beforeEach(() => {
  jest.clearAllMocks();
  OrderByWrapper = mountOrderByWrapper();
  OrderByComponent = OrderByWrapper.find('OrderBy');
  InputElement = OrderByComponent.find('input');
});
afterEach(() => {
  OrderByWrapper.unmount();
});

describe('Filter Component', () => {
  it('Renders correctly', () => {
    expect(toJson(OrderByWrapper)).toMatchSnapshot();
    expect(OrderByComponent.find('.order-by')).toHaveLength(1);
  });
  describe('Texts', () => {
    it('By default appears "Newest" Text', () => {
      const labelWrapper = OrderByComponent.find('.p-on');
      const labelElement = labelWrapper.find('label');
      expect(labelElement).toHaveLength(1);
      expect(labelElement.text()).toBe('filter.newest-first');
      expect(labelWrapper).toHaveStyleRule('opacity', '1');
      // expect(labelWrapper.styleRule('opacity').toBe('1'));
    });
    it('After first click appears "Newest" Text', () => {
      OrderByComponent.find('input').simulate('change', { target: { checked: true } });
      const labelWrapper = OrderByComponent.find('.p-off');
      const labelElement = labelWrapper.find('label');
      expect(labelElement).toHaveLength(1);
      expect(labelElement.text()).toBe('filter.latest-first');
      expect(labelWrapper.prop('style')).toHaveProperty('opacity', '1');
    });
  });

  describe('Checked Values', () => {
    it('Input unchecked by default', () => {
      expect(InputElement).toHaveLength(1);
      expect(InputElement.getElement().props.checked).toBe(false);
    });
    it('In the first click change order to ASC', () => {
      OrderByComponent.find('input').simulate('change', { target: { checked: true } });
      expect(defaultProps.changeOrderBy).toHaveBeenCalledTimes(1);
      expect(defaultProps.changeOrderBy.mock.calls[0][0]).toBe('handle_id_ASC');
    });
    it('In the second click change order to DESC', () => {
      OrderByComponent.find('input').simulate('change', { target: { checked: true } });
      OrderByComponent.find('input').simulate('change', { target: { checked: false } });
      expect(defaultProps.changeOrderBy).toHaveBeenCalledTimes(2);
      expect(defaultProps.changeOrderBy.mock.calls[1][0]).toBe('handle_id_DESC');
    });
  });
});
