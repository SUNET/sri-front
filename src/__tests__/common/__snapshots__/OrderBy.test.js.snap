// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Filter Component Renders correctly 1`] = `
<CreateOrderByWrapper>
  <div>
    <OrderBy
      changeOrderBy={[MockFunction]}
      t={
        [MockFunction] {
          "calls": Array [
            Array [
              "filter.newest-first",
            ],
            Array [
              "filter.latest-first",
            ],
          ],
          "results": Array [
            Object {
              "type": "return",
              "value": "filter.newest-first",
            },
            Object {
              "type": "return",
              "value": "filter.latest-first",
            },
          ],
        }
      }
    >
      <div
        className="order-by outline"
      >
        <FieldSwitch
          classNames="icon-right"
          handleChecked={[Function]}
          icon="angle"
          id="orderBy"
          labelChecked="filter.newest-first"
          labelUnChecked="filter.latest-first"
          type="toggle-icon"
        >
          <div
            className="field-checkbox"
          >
            <div
              className="pretty custom p-icon p-toggle p-plain icon-right"
            >
              <input
                checked={false}
                id="orderBy"
                onChange={[Function]}
                type="checkbox"
              />
              <div
                className="state p-on undefined"
              >
                <label>
                  filter.newest-first
                </label>
                <FontAwesomeIcon
                  border={false}
                  className=""
                  fixedWidth={false}
                  flip={null}
                  icon={
                    Object {
                      "icon": Array [
                        320,
                        512,
                        Array [],
                        "f106",
                        "M177 159.7l136 136c9.4 9.4 9.4 24.6 0 33.9l-22.6 22.6c-9.4 9.4-24.6 9.4-33.9 0L160 255.9l-96.4 96.4c-9.4 9.4-24.6 9.4-33.9 0L7 329.7c-9.4-9.4-9.4-24.6 0-33.9l136-136c9.4-9.5 24.6-9.5 34-.1z",
                      ],
                      "iconName": "angle-up",
                      "prefix": "fas",
                    }
                  }
                  inverse={false}
                  listItem={false}
                  mask={null}
                  pull={null}
                  pulse={false}
                  rotation={null}
                  size={null}
                  spin={false}
                  swapOpacity={false}
                  symbol={false}
                  title=""
                  transform={null}
                >
                  <svg
                    aria-hidden="true"
                    className="svg-inline--fa fa-angle-up fa-w-10 "
                    data-icon="angle-up"
                    data-prefix="fas"
                    focusable="false"
                    role="img"
                    style={Object {}}
                    viewBox="0 0 320 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M177 159.7l136 136c9.4 9.4 9.4 24.6 0 33.9l-22.6 22.6c-9.4 9.4-24.6 9.4-33.9 0L160 255.9l-96.4 96.4c-9.4 9.4-24.6 9.4-33.9 0L7 329.7c-9.4-9.4-9.4-24.6 0-33.9l136-136c9.4-9.5 24.6-9.5 34-.1z"
                      fill="currentColor"
                      style={Object {}}
                    />
                  </svg>
                </FontAwesomeIcon>
              </div>
              <div
                className="state p-off"
              >
                <label>
                  filter.latest-first
                </label>
                <FontAwesomeIcon
                  border={false}
                  className=""
                  fixedWidth={false}
                  flip={null}
                  icon={
                    Object {
                      "icon": Array [
                        320,
                        512,
                        Array [],
                        "f107",
                        "M143 352.3L7 216.3c-9.4-9.4-9.4-24.6 0-33.9l22.6-22.6c9.4-9.4 24.6-9.4 33.9 0l96.4 96.4 96.4-96.4c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9l-136 136c-9.2 9.4-24.4 9.4-33.8 0z",
                      ],
                      "iconName": "angle-down",
                      "prefix": "fas",
                    }
                  }
                  inverse={false}
                  listItem={false}
                  mask={null}
                  pull={null}
                  pulse={false}
                  rotation={null}
                  size={null}
                  spin={false}
                  swapOpacity={false}
                  symbol={false}
                  title=""
                  transform={null}
                >
                  <svg
                    aria-hidden="true"
                    className="svg-inline--fa fa-angle-down fa-w-10 "
                    data-icon="angle-down"
                    data-prefix="fas"
                    focusable="false"
                    role="img"
                    style={Object {}}
                    viewBox="0 0 320 512"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M143 352.3L7 216.3c-9.4-9.4-9.4-24.6 0-33.9l22.6-22.6c9.4-9.4 24.6-9.4 33.9 0l96.4 96.4 96.4-96.4c9.4-9.4 24.6-9.4 33.9 0l22.6 22.6c9.4 9.4 9.4 24.6 0 33.9l-136 136c-9.2 9.4-24.4 9.4-33.8 0z"
                      fill="currentColor"
                      style={Object {}}
                    />
                  </svg>
                </FontAwesomeIcon>
              </div>
            </div>
          </div>
        </FieldSwitch>
      </div>
    </OrderBy>
  </div>
</CreateOrderByWrapper>
`;
